crates/sui-core/src/traffic_controller/nodefw_client.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021, Facebook, Inc. and its affiliates
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BlockAddresses {
+    pub addresses: Vec<BlockAddress>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BlockAddress {
+    pub source_address: String,
+    pub destination_port: u16,
+    pub ttl: u64,
+}
+
+pub struct NodeFWClient {
+    client: reqwest::Client,
+    remote_fw_url: String,
+}
+
+impl NodeFWClient {
+    pub fn new(remote_fw_url: String) -> Self {
+        Self {
+            client: reqwest::Client::new(),
```

Comment:
it would be nice to have the client injected for unit testing, when the time comes.

Reply:
I went with a different approach of building a test server rather than allowing for injecting the client. Seemed easier to do that than to build a mock reqwest client + generics to support it. You could argue that client mocking would allow us to remove the possibility of network flakiness from a real server running, but even if we did that, we have a real network running in these integration tests for json rpc and grpc